wolfcrypt/src/pkcs12.c

```diff
@@ -670,6 +670,33 @@ int wc_d2i_PKCS12(const byte* der, word32 derSz, WC_PKCS12* pkcs12)
     return ret;
 }
 
+/* Convert WC_PKCS12 struct to allocated DER buffer.
+ * pkcs12 : non-null pkcs12 pointer
+ * der    : pointer-pointer to der buffer holding PKCS12
+ * return size of DER on success and negative on failure.
+ */
+int wc_i2d_PKCS12(WC_PKCS12* pkcs12, byte** der)
+{
+    int ret;
+    byte *buf = NULL;
+
+    if (pkcs12 == NULL) {
+        ret = BAD_FUNC_ARG;
+    }
+    else {
+        buf = (byte*)XMALLOC(pkcs12->safe->dataSz, pkcs12->heap,
+                DYNAMIC_TYPE_PKCS);
+        if (buf == NULL) {
+            ret = MEMORY_E;
+        }
+        else {
+            XMEMCPY(buf, pkcs12->safe->data, pkcs12->safe->dataSz);
```

Comment:
This buffer does not quite contain all that is needed for a DER format PKCS12 buffer. It contains a safe bag type (which can be safeContent type with multiple nested bags). When out putting the DER this is missing the version and if added the optional macData.

Reply:
Check